### DISCUSSION TITLE
Teslamate integration - Tesla Vehicles

### DISCUSSION THREAD

**Discussion by radh21301** [Freika/dawarich#248] [Ideas]
I have [Teslamate](https://github.com/teslamate-org/teslamate) running which continuously polls the location of the Tesla vehicle (among so many other things). It would be really nice if Dawarich can pull data from Teslamate. I can then stop logging on my phone and save a lot of battery. And honestly. most of my travels are in the car, so this essentially tracks everything in Dawarich.

Looking at the number of projects in the [Community Projects](https://docs.teslamate.org/docs/projects) page on Teslamate docs, it feels like the api is quite extensive which might make it easy for Dawarich to integrate with it. But that is just my non-developer mind talking.

**Comment by Freika** [comment]
Sounds interesting, I'll look into it!

**Comment by Freika** [comment]
@radh21301 is there a chance you could share some data from your mqtt instance so I could experiment with integration?

**Reply by radh21301** [reply]
Hey. I am not entirely sure how to do that. But I copied the data that the mqtt broker has right now, and pasted them in a text file. Maybe there is a better way to do this or you are looking for something else?
[teslamate_mqtt_data.txt](https://github.com/user-attachments/files/17726170/teslamate_mqtt_data.txt)

This is the link to the yaml file that I used in homeassistant to read values from the mqtt broker - maybe this helps somehow.
https://docs.teslamate.org/docs/integrations/home_assistant#mqtt_sensorsyaml-mqtt-section-of-configurationyaml

**Reply by Freika** [reply]
I'll have a look and will return to you :)
